### PR TITLE
IPS-1098: Implement Lambda canary deployment for BAV CRI

### PR DESCRIPTION
**Pull request by madankaruppiah**
- Added LambdaCanaryDeployment preference on updated the pipeline
    - Added Code Deploy Service role to allow code deploy
    - Set LambdaDeploymentPreference to AllAAatOnce
    - Updated lambda functions with the canary deployment configurations
    - Added additional alarms for Lambda errors and 5xx errors with lower threshold

## Proposed changes
- Implement Lambda canary deployment for BAV CRI 
### What changed
- Implement Lambda canary deployment for BAV CRI 

### Issue tracking

- [KIWI-1098](https://govukverify.atlassian.net/browse/KIWI-1098)

### PII logging

- [ ] Verified that no PII data is being logged

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

Screenshot 2024-09-10 at 16 39 12

Screenshot 2024-09-11 at 14 29 26

[KIWI-1098]: https://govukverify.atlassian.net/browse/KIWI-1098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: deploy/template.yaml
[the remark this answers is not in the document]
Comment (madankaruppiah) [reply deploy/template.yaml:1]: @louiseli1 , working with dev to fix the infra tests.

[the remark this answers is not in the document]
Comment (madankaruppiah) [reply deploy/template.yaml:1]: Fixed the circular dependency by changing the alarm dimensions.

File: deploy/template.yaml [hunk 1]
```diff
@@ -876,6 +886,63 @@ Resources:
...
+    Type: AWS::CloudWatch::Alarm
+    Condition: UseCanaryDeploymentAlarms
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - !ImportValue platform-alarm-topic-slack-warning-alert
+      OKActions:
+        - !ImportValue platform-alarm-topic-slack-warning-alert
+      AlarmName: !Sub "${AWS::StackName}-SessionFunctionCanary5xxErrors"
+      AlarmDescription: !Sub "SessionFunction Lambda returning 5xx response."
```
Comment (louiseli1) [inline deploy/template.yaml:925]: Add support manual link? (Or remove `!Sub`, same for other AlarmDescriptions)

File: deploy/template.yaml
Comment (louiseli1) [inline deploy/template.yaml:1]: There are some commented sections for canaries, e.g HmrcTokenFunction - are these needed?

File: deploy/template.yaml (hunk 1 above)
Comment (madankaruppiah) [reply deploy/template.yaml:925]: Done

**Review by GDainton — APPROVED** [review]

**Pull request merged**